datasets/math_dataset/math_dataset.py

```diff
@@ -232,9 +232,9 @@ def _read_data_from_all_categories(self, directory, config, categories):
                 with open(data_file, encoding="utf-8") as f:
                     ls = f.read().split("\n")
 
-                    for l in ls[::-1]:
-                        if not l:
-                            ls.remove(l)
+                    for k in ls[::-1]:
```

Comment:
I guess you can replace "l" by "line" instead of "k"

Reply:
Sure, done @lhoestq